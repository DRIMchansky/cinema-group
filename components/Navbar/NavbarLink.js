// packages
import styled from 'styled-components'
import Link from 'next/link'
// utils
import { white, primary } from '../../utils/colors'

const NavbarLink = ({ name, href, type }) => {
  return (
    <Link href={href} passHref>
      <StyledNavbarLink type={type}>{name}</StyledNavbarLink>
    </Link>
  )
}

const StyledNavbarLink = styled.a`
  display: block;
  color: ${white};
  text-decoration: none;
  font-weight: 500;
  font-size: ${(props) => (props.type === 'primary' ? '1.2rem' : '1rem')};
  padding: 0.3rem 0.5rem;
  width: 100%;

  &:hover {
  }
`

export default NavbarLink
