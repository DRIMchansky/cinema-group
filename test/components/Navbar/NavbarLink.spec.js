import React from 'react'
import { mount } from 'enzyme'
import NavbarLink from '../../../components/Navbar/NavbarLink'

test('NavbarLink renders correctly', () => {
  const wrapper = mount(<NavbarLink href="/" />)

  expect(wrapper).toMatchSnapshot()
})
