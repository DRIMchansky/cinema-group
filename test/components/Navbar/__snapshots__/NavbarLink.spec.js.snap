// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink renders correctly 1`] = `
.c0 {
  display: block;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 500;
  font-size: 1.3rem;
  padding: 0.7rem 0 0.3rem 0.5rem;
  width: 100%;
}

@media (min-width:960px) {
  .c0 {
    font-weight: 400;
    font-size: 1.1rem;
    padding: 0.5rem;
  }

  .c0:hover {
    opacity: 0.8;
  }
}

<NavbarLink
  href="/"
>
  <Link
    href="/"
    passHref={true}
  >
    <NavbarLink__StyledNavbarLink
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <a
        className="c0"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      />
    </NavbarLink__StyledNavbarLink>
  </Link>
</NavbarLink>
`;
